### PR TITLE
Feature: Docker multiarch image (amd64 & arm64) dynamic tags and ghcr repositoy

### PR DESCRIPTION
**Pull request by sweisgerber**
Hello,

this pull-request targets:

- Fixes: https://github.com/silverbulletmd/silverbullet/issues/136
- Extends: https://github.com/silverbulletmd/silverbullet/issues/198

Additionally it also pushes to the github container registry (GHCR.io)

An Example registry deployment can get found for the imaginary next release:

- https://github.com/sweisgerber/silverbullet/releases/tag/0.2.14
    - DockerHub: https://hub.docker.com/layers/sweisgerber/silverbullet/0.2.14/images/sha256-81ec535cb4b71a01623f15aa93587af6ef6639254478d91f794d986bc829fc3f?context=explore
    - ghcr.io: https://github.com/sweisgerber/silverbullet/pkgs/container/silverbullet/91495509?tag=0.2.14

If desired I can rollback pushing to ghcr or add more tags (e.g. for branches).

### CODE REVIEW

**Comment by zefhemel** [comment]
Oh this is excellent, I will merge this to see if this all works.

**Comment by zefhemel** [comment]
Confirmed: https://hub.docker.com/r/zefhemel/silverbullet/tags there are now ARM and Intel builds, thanks a lot!

**Comment by zefhemel** [comment]
A tagged release will happen for the next release I suppose

**Comment by sweisgerber** [comment]
@zefhemel yes, that's right.

If you would like to have different semantics, like, e.g.
- latest for most recent tagged version and
- edge for the most recent build on master 

I could help with that as well. I think it's a matter of preference. :)

**Comment by ahbnr** [comment]
Is it possible that switching the base image for deno also caused the user id to change?
Because my instance started crashing on startup since the update from yesterday with
```
Permission denied (os error 13): open '/space/data.db'
```

The files in my silverbullet volume all have the permissions 1000:1000.
However, executing `id` inside the container now returns
```
uid=1993(deno) gid=1993(deno) groups=1993(deno)
```

So its uid 1993 now, not 1000 [as in the old base image](https://github.com/silverbulletmd/silverbullet/blob/c8d0b2df146742d7e07d04f6eb8e647ad939c301/Dockerfile#L33).

Backing up all files and destroying the volume fixes it, because the re-created volume automatically gets the new uid 1993.

**Comment by zefhemel** [comment]
Oh that's not good. I think a fair number of people are using this image and will have uid 1000 as the owner of their files. Yeah I think the best would be to add a `silverbullet` user with uid 1000 and use that.

**Comment by sweisgerber** [comment]
Allright, I'll fix it shortly

**Comment by sweisgerber** [comment]
Created issue: https://github.com/silverbulletmd/silverbullet/issues/402 @ahbnr

**Comment by zefhemel** [comment]
@ahbnr the fix has now been merged and a new image pushed, can you confirm this resolves the issue?

**Comment by ahbnr** [comment]
Sorry for the late reply!  
Thank  you very much for handling this issue so quickly, @sweisgerber @zefhemel .

I can not test the new image on the original volume where the error occurred, because I already deleted that one and recreated it with the 1993 owner permissions.
However, I have repeated the process of deleting the volume and recreating the files under the ownership of 1000:1000.
With this setup, I switched to the new image and it is working.

In short, the container seems to be running with the user 1000:1000 again, and I do not encounter any issues when the volume and the files are owned by 1000:1000.

So I think the fix is working.
